src/SFA.DAS.Tools.Servicebus.Support.Web/Controllers/MessageListController.cs

```diff
@@ -93,21 +93,23 @@ public async Task<IActionResult> Index()
                     QueueName = queueName
                 })).QueueInfo,
                 UserSession = await _userSessionService.GetUserSession()
-
             });
         }
 
-        public async Task<IActionResult> ReceiveMessages(string queue)
+        public async Task<IActionResult> ReceiveMessages(string data)
         {
-            HttpContext.Session.SetString("queueName", queue);
+            var request = JsonConvert.DeserializeObject<GetMessagesRequest>(data);
+
+            HttpContext.Session.SetString("queueName", request.Queue);
             var count = (await _getQueueMessageCountQuery.Handle(new GetQueueMessageCountQuery()
             {
-                QueueName = queue
+                QueueName = request.Queue
             })).Count;
 
-            await _retrieveMessagesService.GetMessages(queue, count);
-
-            return RedirectToAction("Index");
+            await _retrieveMessagesService.GetMessages(request.Queue, count, request.Qty);
+            
+            //RedirectToAction("Index");
```

Comment:
Guessing we can lose the redirect? Is is now a return Json None because it only adds the messages to the database behind the scenes. What triggers the refresh on the message list page once there is more data?

Reply:
Action is now triggered by javascript sendRequest, that function also takes care of location.reload(); I had to remove redirect to make it work with sendRequest

Reply:
modified